src/public_parameters/mem_cache.rs

```diff
@@ -73,7 +73,10 @@ impl PublicParamMemCache {
                     Ok(Arc::new(pp.clone())) // this clone is VERY expensive
                 }
                 Err(e) => {
-                    eprintln!("{e}");
+                    if matches!(e, Error::CacheError(_)) {
```

Comment:
1. What does the `matches` macro resolves to? Is there a nearby construct a few lines away in this diff within which you could fold this check?
2. is it ok to not print anything if we encounter an error that is not a `CacheError`? In that case, is it ok to continue?